source/dotnet/Library/AdaptiveCards.Rendering.Wpf/AdaptiveContainerRenderer.cs

```diff
@@ -107,44 +112,53 @@ public static void AddContainerElements(Grid uiContainer, IList<AdaptiveElement>
                     {
                         uiContainer.RowDefinitions.Add(new RowDefinition() { Height = GridLength.Auto });
                         Grid.SetRow(uiElement, uiContainer.RowDefinitions.Count - 1);
-
-                        RendererUtil.ApplyIsVisible(uiElement, cardElement);
-                        Grid.SetRow(uiElement, uiContainer.RowDefinitions.Count - 1);
                         uiContainer.Children.Add(uiElement);
+
+                        context.SetVisibility(uiElement, cardElement.IsVisible, tag);
                     }
                     else
                     {
+                        RowDefinition rowDefinition = new RowDefinition() { Height = new GridLength(1, GridUnitType.Star) };
```

Comment:
>RowDefinition [](start = 24, length = 13)

add comment for clarification on what tag information is being stored here and why